subprojects/native/src/main/java/org/gradle/internal/nativeintegration/services/NativeServices.java

```diff
@@ -80,22 +81,70 @@
     private static final NativeServices INSTANCE = new NativeServices();
     private static final JansiBootPathConfigurer JANSI_BOOT_PATH_CONFIGURER = new JansiBootPathConfigurer();
     private static boolean initialized;
+    private static final EnumSet<NativeFeatures> INITIALIZED_FEATURES = EnumSet.noneOf(NativeFeatures.class);
 
     public static final String NATIVE_DIR_OVERRIDE = "org.gradle.native.dir";
 
+    public enum NativeFeatures {
+        FILE_SYSTEM_WATCHING {
+            @Override
+            public void initialize(File nativeBaseDir) {
+                if (useNativeIntegrations) {
```

Comment:
Can this `if` be moved to where we create the enum set instead?

I find it hard to follow the value of `useNativeIntegrations`, maybe if it wasn't overwritten it would be easier to follow the logic?

Reply:
I now pass in the `useNativeIntegrations` into this method. I think it is easier to understand now.